src/foundry/client/pixi/webgl/shaders/filters/filters.d.ts

```diff
@@ -0,0 +1,460 @@
+export {};
+
+declare global {
+  /**
+   * This class defines an interface for masked custom filters
+   */
+  class AbstractBaseMaskFilter extends AbstractBaseFilter {
+    static create(uniforms?: AbstractBaseShader.Uniforms): AbstractBaseMaskFilter;
+
+    /**
+     * The default vertex shader used by all instances of AbstractBaseMaskFilter
+     */
+    static vertexShader: string;
+
+    override apply(
+      filterManager: PIXI.FilterSystem,
+      input: PIXI.RenderTexture,
+      output: PIXI.RenderTexture,
+      clear: PIXI.CLEAR_MODES,
+      currentState: any,
+    ): void;
+  }
+
+  /**
+   * A filter used to control channels intensity using an externally provided mask texture.
+   * The mask channel used must be provided at filter creation.
+   * Contributed by SecretFire#4843
+   */
+  class InverseOcclusionMaskFilter extends AbstractBaseMaskFilter {
+    static create<T extends InverseOcclusionMaskFilter>(
+      this: ConstructorOf<T>,
+      uniforms?: AbstractBaseShader.Uniforms,
+    ): T;
+
+    static adaptiveFragmentShader(channel: "r" | "g" | "b"): string;
+
+    /**
+     * @defaultValue
+     * ```js
+     * {
+     *  uMaskSampler: null,
+     *  alphaOcclusion: 0,
+     *  alpha: 1,
+     *  depthElevation: 0
+     * }
+     * ```
+     */
+    static override defaultUniforms: AbstractBaseShader.Uniforms;
+  }
+
+  /**
+   * A filter used to apply a reverse mask on the target display object.
+   * The caller must choose a channel to use (alpha is a good candidate).
+   */
+  class ReverseMaskFilter extends AdaptiveFragmentChannelMixin(AbstractBaseMaskFilter) {
+    static override adaptiveFragmentShader(channel: "r" | "g" | "b"): string;
+
+    /**
+     * @defaultValue
+     * ```js
+     * { uMaskSampler: null }
+     * ```
+     */
+    static defaultUniforms: AbstractBaseShader.Uniforms;
+  }
+
+  /**
+   * A minimalist filter (just used for blending)
+   */
+  class VoidFilter extends AbstractBaseFilter {
+    static create<T extends VoidFilter>(this: ConstructorOf<T>, uniforms?: AbstractBaseShader.Uniforms): T;
+
+    static override fragmentShader: string;
+  }
+
+  namespace VisualEffectsMaskingFilter {
+    type PostProcessModes = Array<keyof (typeof VisualEffectsMaskingFilter)["POST_PROCESS_TECHNIQUES"]>;
+
+    type FilterMode = ValueOf<(typeof VisualEffectsMaskingFilter)["FILTER_MODES"]>;
+
+    /**
+     * @privateRemarks Implementation taken from https://stackoverflow.com/a/61434547
+     */
+    type CreateOptionsIntersection = {
+      filterMode?: FilterMode;
+      postProcessModes?: PostProcessModes;
+    } & AbstractBaseShader.Uniforms;
```

Comment:
I would put a TODO on this, linking to https://github.com/microsoft/TypeScript/issues/17867. If that issue ever gets resolved the type here can get better. The better we keep track of issues like this the better. If you could do that everywhere you've run into problems like this (e.g. the `(sprite: SpriteMesh)` constructor) that'd be great.